### PR TITLE
Add directory sdk

### PR DESCRIPTION
**Pull request by gimmyxd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +21,117 @@ Or install it yourself as:
...
+- `url`: hostname:port of directory service (_required_)
+- `api_key`: API key for directory service (_required_ if using hosted directory)
+- `tenant_id`: Aserto tenant ID (_required_ if using hosted directory)
+- `cert_path`: Path to the grpc service certificate when connecting to local topaz instance.
+
+### Getting objects and relations
+Get an object instance with the type `type-name` and the key `object-key`. For example:
+
+```ruby
+user = directory_client.object(type: 'user', key: 'euang@acmecorp.com')
```
Comment (oanatmaria) [inline README.md:51]: the `type` should be an `identity` in this case

Comment (gimmyxd) [reply README.md:51]: Not necessarily,  as acmecorp users is just a set modeled by us, afaik the email will be promoted as `key`

File: lib/aserto/directory/client.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+      # Creates a new Directory Client
+      #
+      # @param url [String] the gRpc url of the directory server
+      # @param api_key [String] the api key of the directory server(for hosted directory)
+      # @param tenant_id [String] the tenant id of the directory server(for hosted directory)
+      # @param cert_path [String] the path to the certificates folder
+      #
+      # @return [Aserto::Directory::Client] the new Directory Client
+      def initialize(url: "directory.prod.aserto.com:8443", api_key: nil, tenant_id: nil, cert_path: nil)
```
Comment (oanatmaria) [inline lib/aserto/directory/client.rb:20]: should we default the url to the one set in topaz `https://localhost:9292` as we did for dotnet https://github.com/aserto-dev/aserto-dotnet/blob/3bb2a453533161d42ea0321ec94bc98f047c0b73/src/Aserto.AspNetCore.Middleware/Options/AsertoOptionsDefaults.cs#L38

**Review by oanatmaria — APPROVED** [review]

File: lib/aserto/directory/client.rb (hunk 2 above)
Comment (gimmyxd) [reply lib/aserto/directory/client.rb:20]: node defaults to prod, we should decide on a value and use it in all the SDKs: https://github.com/aserto-dev/aserto-node/blob/main/lib/ds.ts#L64

**Pull request merged**